common/context/context.cpp

```diff
@@ -53,6 +65,52 @@ kv_db::KeyValueDatabase &MenderContext::GetMenderStoreDB() {
 	return mender_store_;
 }
 
+ExpectedProvidesData MenderContext::LoadProvides() {
+	auto ex_bytes = mender_store_.Read(provides_key_);
+	if (!ex_bytes) {
+		if (ex_bytes.error().code == kv_db::MakeError(kv_db::KeyError, "").code) {
+			// no such data is not an error, just return an empty result
+			ProvidesData ret {};
+			return ExpectedProvidesData(ret);
+		} else {
+			// everything else is an error
+			return ExpectedProvidesData(expected::unexpected(ex_bytes.error()));
```

Comment:
I know you are a fan of explicit construction, but I must admit I find these harder to read. It's harder to visually distinguish good paths from error paths, because they look more alike. But it's not wrong of course, just my 2 cents.

Reply:
Yeah, I have mixed feelings about this myself.